daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1578,47 +1579,81 @@ object Main extends Logging {
     1
   }
 
+  object ExitCode extends Enumeration {
+
+
+
+
+    val Normal = Value(0)
+    val FileNotFound = Value(1)
+    val InvalidParserException = Value(2)
+    val ExternalVariableException = Value(3)
+    val BindingException = Value(4)
+    val NotYetImplemented = Value(5)
+    val TDMLException = Value(6)
+    val OutOfMemory = Value(7)
+    val UserDefinedFunctionFatalError = Value(8)
+    val BugFound = Value(9)
+
+
+    val ParseError = Value(20)
+    val UnparseError = Value(30)
+    val NoForwardProgress = Value(21)
+    val LeftOverData = Value(22)
+    val NotEnoughData= Value(23)
+
+    val SchemaDefinitionError = Value(40)
+    val UnableToCreateProcessor = Value(41)
+    val UnableToGenerateCode = Value(42)
+
+    val PerformanceTestFailed = Value(50)
+  }
+
   def main(arguments: Array[String]): Unit = {
+
     val ret = try {
       run(arguments)
     } catch {
       case s: scala.util.control.ControlThrowable => throw s
       case e: java.io.FileNotFoundException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.FileNotFound
       }
       case e: InvalidParserException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.InvalidParserException
       }
       case e: ExternalVariableException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.ExternalVariableException
       }
       case e: BindingException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.BindingException
       }
       case e: NotYetImplementedException => {
         nyiFound(e)
+        ExitCode.NotYetImplemented
       }
       case e: TDMLException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.TDMLException
       }
       case e: OutOfMemoryError => {
         oomError(e)
+        ExitCode.OutOfMemory
       }
       case e: UserDefinedFunctionFatalErrorException => {
         log(LogLevel.Error, "%s", e.getMessage())
         e.printStackTrace()
-        1
+        ExitCode.UserDefinedFunctionFatalError
```

Comment:
Suggest renaming this to UserDefinedFunctionError. The "Fatal" part is a way for Daffodil to differentiate between different kinds of UDF errors. But by the time we get to the CLI, the only UDF error that a CLI user could ever see is the fatal error. So all UDF errors are implied to be fatal from a CLI perspective.